add documentation for OTB provider (#5783)
qgis/QGIS-Documentation: docs/user_manual/processing_algs/otb/index.rst

*************************
OTB applications provider
*************************

> `OTB <https://www.orfeo-toolbox.org>`_ (Orfeo ToolBox) is image  

Reviewer: ```suggestion
`OTB <https://www.orfeo-toolbox.org>`_ (Orfeo ToolBox) is an image  
```

> processing library for remote sensing data. It also provide a set of 

Reviewer: ```suggestion
processing library for remote sensing data. It also provides
```

> applications which provides a set of image processing functionnalities. 

Reviewer: ```suggestion
applications that provide image processing functionalities. 
```

> The list and the documentation of this applications are available in 

Reviewer: ```suggestion
The list of applications and their documentation are available in
```

> `OTB CookBook <https://www.orfeo-toolbox.org/CookBook/Applications.html>`_

Reviewer 2: Based on what I mention above, I'm not sure this description fits what this chapter is dedicated to (ie list the available algs of the provider). This info is for the third-party file.

Author: As comment below the documentation of QGIS Processing providers must be refactored because it is not clear for me where put the information about OTB application. If you check tauDem page you will see that a part of the content must be put in configuration.

Reviewer: @savmickael, I think @DelazJ's suggestion to move this introduction to 3rdparty.rst is sensible. If you agree, I suggest you try to integrate it with what is already there.

Author: done in 1195c5c

Reviewer 2: > If you check tauDem page you will see that a part of the content must be put in configuration.

If you refer to https://docs.qgis.org/3.10/en/docs/user_manual/processing_algs/taudem/index.html (the only place taudem is mentioned afaics), IMO it's more an introduction to the software whose algs are about to be described. And I'm not sure it's worth splitting  for one or two sentences. Also ref my understanding of the configuration file purpose

Reviewer: Since processing_algs in the User Manual is where users go to find algorithm documentation, I find it sensible to have a pointer to the OTB algorithm / application documentation here. No problem spending a page for that link, but if we can come up with a better solution - no problem.
